### ISSUE TITLE
Option to let version_file take precedence

### ISSUE THREAD

**Issue by tomas16** [dolfinus/setuptools-git-versioning#97]
Context: I'd like to define a "release version" in a file, and then use git versioning for commits in between official releases. Our package generates data and for reproducibility we access the full version number at runtime and save it along with the data.

The [documentation about version_file](https://setuptools-git-versioning.readthedocs.io/en/stable/schemas/file/version_file.html) says:

> Please take into account that version_file is ignored if any tag is present in the current branch.

This creates the following problems:
* Our repo initially only had a main branch, and as it grew we added a develop branch and have a lightweight form of git-flow going on. Since develop branched from the main branch, which already had some version tags, the version_file is ignored for all eternity.
* Even if that weren't the case, if we update the version file on develop and then merge into main, the version file is ignored again. Even though tags are really just for convenience and the ground truth release version is contained in a file, if we forget to add the tag on main, the version reverts to the previous tag.

I looked into using version_callback instead. I tried adding a `__release_version__ = "0.1.0"` string inside `__init__.py` and in a separate `_version.py` file. The respective entries in `pyproject.toml` I tried are `version_callback = "mypackage:__release_version__"` and `version_callback = "mypackage._version:__release_version__"`

However I ran into some issues with this as well:
* I can't get setuptools-git-versioning to actually use this string. I'm getting `packaging.version.InvalidVersion: Invalid version: 'mypackage._version.__release_version__'` when I either `pip install -e .` or run `python -m setuptools_git_versioning` in a terminal. Strangely, when I run that last command in a debugger it spits out the correct version number.
* Even if I can get past this problem, there's another one I don't see a way around. Since I need the full version number, including any "post" and "git" suffixes at runtime, I'm using the `importlib` method. However, when the package hasn't been installed, I don't ignore the `PackageNotFoundError` and instead tell users to install the package first. The problem is whenever setuptools-git-versioning would try to import the callback, `__init__.py` would get imported as well, and we'd get this `PackageNotFoundError` since we're in the process of installing the package.

Maybe I'm not using this tool right, so I'm open to suggestions.
My request however, would be to add an option to let `version_file` take precedence over any git tags. That would solve the whole problem.

**Comment by dolfinus** [comment]
Hi.

Could you please provide your `pyproject.toml`?

It is possible that these lines are missing in the config:
```toml
[build-system]
# __legacy__ is required to have access to package
# during build step
build-backend = "setuptools.build_meta:__legacy__"

[project]
dynamic = ["version"]
```

**Comment by tomas16** [comment]
I figured out a way to do what I want.... more or less.

`pyproject.toml`:
```toml
[tool.setuptools-git-versioning]
enabled = true
version_file = "VERSION"
count_commits_from_version_file = true
tag_filter = "$^"       # ignore tags
```

The regular expression in `tag_filter` matches nothing, so tags are ignored in favor of the `version_file`.

There's one remaining issue I can live with: this will always use the `dev_template` or `dirty_template`. So even when the version number was updated in the last commit, you get something like `0.1.2.post0+git.b0ce782f` instead of just `0.1.2`. 

According to the documentation, this is expected as `template` is only used if the current commit has been tagged.

So an **updated feature request** could be: "use `template` when `ccount == 0` and there are no uncommitted or untracked changes."


Finally, the `tag_filter` trick kind of contradicts the documentation [here](https://setuptools-git-versioning.readthedocs.io/en/stable/options/tag_filter.html):

> This option is completely ignored if [Read some file content as current version](https://setuptools-git-versioning.readthedocs.io/en/stable/schemas/file/version_file.html#version-file) schema is used. This is because all tags are set on master / main branch, so commits to other branches like develop are tagless.

I guess it's only implicitly ignored, because the assumption is there are no tags in the first place.
